src/utils/walk.ts

```diff
@@ -33,20 +33,28 @@ function assignNodeSpecificFields(base: IBaseNode, node: JSONSchema4) {
   }
 }
 
+function getType(node: JSONSchema4) {
+  if (Array.isArray(node.type)) {
+    return node.type.length === 1 ? node.type[0] : node.type;
```

Comment:
We return a string if there's only one, but an array if theres none or more than one? Is that correct? Just making sure

Reply:
> We return a string if there's only one, but an array if theres none or more than one? Is that correct? Just making sure

Well, I'm not sure how to handle such a scenario. For instance, there is no way we can display both `array` of items and `object` with properties.
Our UI is just too simple.

We would need to have something more or less similar to http://jlblcc.github.io/json-schema-viewer/ to handle such a case.

@marbemac what your thought on that?

Reply:
Also, we should be handling `not`, which we don't at the moment.
Any suggestions on the UI?

Reply:
> there is no way we can display both array of items and object with properties

Is this what you mean?

![image](https://user-images.githubusercontent.com/7423098/57644217-3dd34e00-7581-11e9-9266-2c53b63d6978.png)


Or something like `oneOf`?

![image](https://user-images.githubusercontent.com/7423098/57643901-70307b80-7580-11e9-8590-d9c4a69410c1.png)

Reply:
What is `not`?

We don't have to handle array and object in same property. We can suggest people use `oneOf` if they need that, although its kind of crazy and definitely an edge case.

Reply:
https://json-schema.org/understanding-json-schema/reference/combining.html#not

Reply:
@lottamus yeah, pretty much that
```json
{
  "type": ["array", "object"],
  "properties": {
    "bar": { 
      "type": "string" 
    }
  },
  "items": [
     {
       "type": "number"
     },
     {
       "type": "object",
       "properties": {
          "foo": { 
            "type": "number" 
          }
        }
     }
  ]
}

Reply:
Got it, we can skip `not` for now - I haven't seen it used in the wild by our customers yet, and would need to support it in JSE as well. A good addition for later, but not high priority right now.